common/src/main/java/org/apache/cassandra/sidecar/common/JmxClient.java

```diff
@@ -259,12 +259,16 @@ private Map<String, Object> buildJmxEnv()
     }
 
     @Override
-    public synchronized void close() throws IOException
+    public void close() throws IOException
     {
-        JMXConnector connector = jmxConnector;
-        if (connector != null)
+        JMXConnector connector;
+        synchronized (this)
```

Comment:
Can you make `connect()` (where `jmxConnector` is update) synchronized? It is synchronized in its only callsite. It won't hurt performance when synchronizing `connnect()`, but more error-proof.